RUM-8425 Add SwiftUI Auto-tracking

mariedm opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do? What is the motivation?
<!-- A brief description of the change being made with this pull request. What is your motivation for the PR? -->

This PR adds documentation about SwiftUI Auto-tracking (iOS), which is about to be released in [2.29.0](https://github.com/DataDog/dd-sdk-ios/releases/tag/2.29.0).

### Merge instructions

<!-- 
If you're waiting for a release or there are other considerations that you want us to be aware of, list them here. 
If the PR is ready to be merged once it receives the required reviews, check the box below after you've created the PR.
-->

Merge readiness:
- [x] Ready for merge

**For Datadog employees**:

Your branch name MUST follow the `<name>/<description>` convention and include the forward slash (`/`). Without this format, your pull request will not pass CI, the GitLab pipeline will not run, and you won't get a branch preview. Getting a branch preview makes it easier for us to check any issues with your PR, such as broken links.

If your branch doesn't follow this format, rename it or create a new branch and PR.

[6/5/2025] Merge queue has been disabled on the documentation repo. If you have write access to the repo, the PR has been reviewed by a Documentation team member, and all of the required checks have passed, you can use the **Squash and Merge** button to merge the PR. If you don't have write access, or you need help, reach out in the #documentation channel in Slack.

### Additional notes
<!-- Anything else we should know when reviewing?-->

<!-- Previewing the PR: Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running. -->

rtrieu commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/ios/advanced_configuration.md:
```diff
@@ -476,9 +478,134 @@ class YourCustomPredicate: UIKitRUMViewsPredicate {
+- The implementation differs between iOS 18+ and iOS 17 and below:
+  - **iOS 18 and above:** Most interactions are reliably tracked with correct component names (e.g., `SwiftUI_Button`, `SwiftUI_NavigationLink`).
+  - **iOS 17 and below:** The SDK cannot distinguish between interactive and non-interactive components (for example, Button vs. Label). Actions are reported as `SwiftUI_Unidentified_Element`.
+- You can use the default predicate, `DefaultSwiftUIRUMActionsPredicate`, or provide your own to filter or rename actions. You can also disable legacy detection (iOS 17-) if you only want reliable iOS 18+ tracking:
```
```suggestion
- You can use the default predicate, `DefaultSwiftUIRUMActionsPredicate`, or provide your own to filter or rename actions. You can also disable legacy detection (iOS 17) if you only want reliable iOS 18+ tracking:
```

rtrieu reviewed (changes requested):
Hi @mariedm thank you for this PR! I added some feedback mostly around formatting and to comply with our style guide. Let me know if you have any questions or want to chat through any of this.

mariedm commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/ios/advanced_configuration.md:
```diff
@@ -476,9 +478,134 @@ class YourCustomPredicate: UIKitRUMViewsPredicate {
+- On tvOS, only press interactions on the remote are tracked. Only a UIKit predicate is needed for this. If you have a pure SwiftUI app but want to track remote presses on tvOS, you should also enable UIKit instrumentation.
+- The implementation differs between iOS 18+ and iOS 17 and below:
+  - **iOS 18 and above:** Most interactions are reliably tracked with correct component names (e.g., `SwiftUI_Button`, `SwiftUI_NavigationLink`).
+  - **iOS 17 and below:** The SDK cannot distinguish between interactive and non-interactive components (for example, Button vs. Label). Actions are reported as `SwiftUI_Unidentified_Element`.
```
We never can make a distinction, so we **always** report interactions as `SwiftUI_Unidentified_Element`.

mariedm commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/ios/advanced_configuration.md:
```diff
@@ -476,9 +478,134 @@ class YourCustomPredicate: UIKitRUMViewsPredicate {
+<div class="alert alert-info"><p>Note: This feature is experimental and may change in future releases.</p></div>
+
+**Recommendations and notes:**
+- It is recommended to enable UIKit action tracking as well, even if your app is built entirely with SwiftUI. Many interactive components are implemented using UIKit.
```
I think it's best to leave "as well" here because we want customers to enable both UIKit and SwiftUI tracking, even if their app is pure SwiftUI.

mariedm commented:
Thanks for the review @rtrieu! I've addressed all your comments and slightly adapted a couple of suggestions. Let me know if additional changes are needed.
